From b26500db37933d9f97d599dcaa4a02a477df6e9b Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Sun, 13 Mar 2011 11:47:37 -0500
Subject: [PATCH 3/9] omap3: beagle: detect new xM revision B

The xM B uses a DM3730 ES1.1 over the ES1.0 on xM A's, no other board changes.

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/mach-omap2/board-omap3beagle.c |    9 +++++++--
 1 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-omap2/board-omap3beagle.c b/arch/arm/mach-omap2/board-omap3beagle.c
index 8672dc9..ebb90dc 100644
--- a/arch/arm/mach-omap2/board-omap3beagle.c
+++ b/arch/arm/mach-omap2/board-omap3beagle.c
@@ -62,7 +62,8 @@
  *	AXBX	= GPIO173, GPIO172, GPIO171: 1 1 1
  *	C1_3	= GPIO173, GPIO172, GPIO171: 1 1 0
  *	C4	= GPIO173, GPIO172, GPIO171: 1 0 1
- *	XM	= GPIO173, GPIO172, GPIO171: 0 0 0
+ *	XMA	= GPIO173, GPIO172, GPIO171: 0 0 0
+ *	XMB	= GPIO173, GPIO172, GPIO171: 0 0 1
  */
 enum {
 	OMAP3BEAGLE_BOARD_UNKN = 0,
@@ -121,7 +122,11 @@ static void __init omap3_beagle_init_rev(void)
 		omap3_beagle_version = OMAP3BEAGLE_BOARD_C4;
 		break;
 	case 0:
-		pr_info("OMAP3 Beagle Rev: xM\n");
+		pr_info("OMAP3 Beagle Rev: xM A\n");
+		omap3_beagle_version = OMAP3BEAGLE_BOARD_XM;
+		break;
+	case 1:
+		pr_info("OMAP3 Beagle Rev: xM B\n");
 		omap3_beagle_version = OMAP3BEAGLE_BOARD_XM;
 		break;
 	default:
-- 
1.7.1

